test/rules/avoid_void_async.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_void_async`
+import 'dart:async';
+
+void main() {
+  () {}; // OK
+  () => null; // OK
+  () async {}; // OK
+  () async => null; // OK
+}
+
+void a() async {} // LINT
+void b() {} // OK
+Future<void> c() async {} // OK
+void d() async => null; // LINT
+void e() => null; // OK
+Future<void> f() async => null; // OK
+
+void g() async* {} // LINT
+void h() sync* {} // LINT
+Stream<void> i() async* {} // OK
+Iterable<void> j() sync* {} // OK
+
+void get k => null; // OK
```

Comment:
I'm mostly kidding, but should we have a lint for getters whose return type is `void`? Seems kind of pointless if you follow the philosophy that getters should not have side-effects. :-)

Reply:
It would be a good case for an easter egg...generate an "error, expected: bed time, but got: late night code that won't pass review in the morning" :)